docs/DeveloperGuide.md

```diff
@@ -36,3 +92,315 @@
+3. Integration with the `ShootDirectionConverter` class to validate the user's input and ensure it's within the expected range.
+
+### Implementation
+
+Here is the implementation structure of the `Save` class:
+
+```java
+public class Save {
+    
+  public static void executeSave() {
+    int userSaveDirection = getUserSaveDirection();
+    int aiPenaltyDirection = Ai.getAiDirection();
+    boolean isGoalSaved = userSaveDirection == aiPenaltyDirection;
+
+
+    Formatter.printSaveResult(isGoalSaved);
+    Ui.roundCount++;
+  }
+  
+  private static int getUserSaveDirection() {
+    int userDirection;
+    do {
+      System.out.print("Enter the direction to save the penalty (0, 1, or 2): ");
+      String directionString = Ui.IN.nextLine().trim();
+      userDirection = ShootDirectionConverter.convertToShootDirection(directionString);
+      if (userDirection == -1) {
+        System.out.println("Invalid direction! Please enter 0, 1, or 2.");
+      }
+    } while (userDirection == -1);
+    return userDirection;
+  }
+  
+}
+```
+
+The `executeSave()` method drives the process, starting by invoking `getUserSaveDirection()` to prompt the user for input. It then retrieves the AI's direction with `Ai.getAiDirection()`. A comparison is made, and the result of the attempted save is passed to the `Formatter` for output. The `roundCount` is incremented at the end of the method, signaling the progression of the game.
+
+The `getUserSaveDirection()` method repeatedly prompts the user for input until a valid direction (0, 1, or 2) is entered. Input validation is offloaded to `ShootDirectionConverter.convertToShootDirection()`, which translates the string input into a numeric direction or returns -1 for invalid input.
+
+### Alternatives Considered
+
+The current implementation could be further modularized by separating input collection and input validation into distinct methods. This would potentially improve the readability and maintainability of the code by delineating clear responsibilities within the class structure. Additionally, using exceptions to handle invalid input could provide a cleaner mechanism for managing input errors and reduce the reliance on sentinel values like `-1`.
+
+## DifficultyLevel Enum
+
+### Overview
+
+The `DifficultyLevel` enum represents the difficulty levels of the game. It has three values: `EASY`, `MEDIUM`, and `HARD`.
+
+### Design
+
+The `DifficultyLevel` enum is a simple enumeration that defines the three difficulty levels of the game.
+
+### Implementation
+
+The `DifficultyLevel` enum is implemented as follows:
+
+```java
+public enum DifficultyLevel {
+    EASY,
+    MEDIUM,
+    HARD
+}
+```
+
+### Alternatives Considered
+
+An alternative to using an enum could be to use integer constants or string constants to represent the difficulty levels. However, enums provide type safety and better code readability, making them a preferred choice for representing a fixed set of values.
+
+## Class Diagrams
```

Comment:
Adding sequence diagrams for classes can be beneficial to visualize the interactions between various entities in your system.